### PR TITLE
Fix: toDataURL is not supported for this content on iPhone 7p

### PR DESCRIPTION
**Pull request by yyycggg**
Re: https://github.com/cocos/3d-tasks/issues/13833

### Changelog
*  Fix: toDataURL is not supported for this content on iPhone 7p

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: pal/system-info/minigame/system-info.ts [hunk 1]
```diff
@@ -97,12 +97,17 @@ class SystemInfo extends EventTarget {
         this.isXR = false;
 
         // init capability
+        const isIOSWechat = WECHAT && this.os === OS.IOS && !minigame.isDevTool;
```
Comment (PPpro) [inline pal/system-info/minigame/system-info.ts]:
```suggestion
        // NOTE: canvas.toDataURL() is not supported on WeChat iOS end
        const isIOSWechat = WECHAT && this.os === OS.IOS && !minigame.isDevTool;
```

It's better to add some comment why we do this bug fix

**Review by PPpro — APPROVED** [review]

File: pal/system-info/minigame/system-info.ts [hunk 2]
```diff
@@ -97,12 +97,18 @@ class SystemInfo extends EventTarget {
         this.isXR = false;
 
         // init capability
+        // NOTE: canvas.toDataURL() is not supported on WeChat iOS end (Found on iPhone 7p)
+        const isIOSWechat = WECHAT && this.os === OS.IOS && !minigame.isDevTool;
```
Comment (minggo) [inline pal/system-info/minigame/system-info.ts]: Why need `!minigame.isDevTool`?

File: pal/system-info/minigame/system-info.ts [hunk 3]
```diff
@@ -97,12 +97,18 @@ class SystemInfo extends EventTarget {
...
-            supportWebp  = false;
+        if (isIOSWechat) {
+            supportWebp = true;
+        } else {
+            try {
+                supportWebp = TEST ? false : _tmpCanvas1.toDataURL('image/webp').startsWith('data:image/webp');
+            } catch (e) {
+                supportWebp  = false;
+            }
         }
```
Comment (minggo) [inline pal/system-info/minigame/system-info.ts]: I think it is better to add a function named `supportsWebp`.

Comment (yyycggg) [reply pal/system-info/minigame/system-info.ts]: OK

File: pal/system-info/minigame/system-info.ts (hunk 2 above)
Comment (yyycggg) [reply pal/system-info/minigame/system-info.ts]: I will delete it.

**Review by minggo — APPROVED** [review]

**Pull request merged**